pkg/controller/workload/workload_cache.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package workload
+
+import (
+	"encoding/binary"
+	"sync"
+
+	workloadapi "kmesh.net/kmesh/api/v2/workloadapi"
+)
+
+var (
+	workloadCache = workloadStore{
+		byUid:  make(map[string]*workloadapi.Workload),
+		byAddr: make(map[NetworkAddress]*workloadapi.Workload),
+	}
+)
+
+type workloadStore struct {
+	byUid  map[string]*workloadapi.Workload
+	byAddr map[NetworkAddress]*workloadapi.Workload
+	mutex  sync.RWMutex
+}
+
+type NetworkAddress struct {
+	Network string
+	Address uint32
+}
+
+func GetWorkloadByUid(uid string) *workloadapi.Workload {
+	workloadCache.mutex.RLock()
+	defer workloadCache.mutex.RUnlock()
+	return workloadCache.byUid[uid]
+}
+
+func GetWorkloadByAddr(networkAddress NetworkAddress) *workloadapi.Workload {
+	workloadCache.mutex.RLock()
+	defer workloadCache.mutex.RUnlock()
+	return workloadCache.byAddr[networkAddress]
+}
+
+func composeNetworkAddress(network string, addr uint32) NetworkAddress {
+	networkAddress := NetworkAddress{
+		Network: network,
+		Address: addr,
+	}
+
+	return networkAddress
+}
+
+func workloadDataToCache(workload *workloadapi.Workload) {
+	uid := workload.Uid
+
+	workloadCache.mutex.Lock()
+	defer workloadCache.mutex.Unlock()
+	_, exist := workloadCache.byUid[uid]
+	if !exist {
+		workloadCache.byUid[uid] = workload
+	}
+
+	for _, ip := range workload.Addresses {
+		addr := binary.LittleEndian.Uint32(ip)
```

Comment:
we have a util function, use that

Reply:
The IP data type here is byte, and existing util function input parameter is a string, or convert to big endian. Previously, we always use little endian formats.

Reply:
ConvertIPByteToUint32 already use big endian

Reply:
recover ConvertIPByteToUint32 to convert ip to little endian, and use it.